apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/navigation-menu/edit.js

```diff
@@ -34,13 +36,37 @@ const NavigationMenuEdit = ( {
 	setTextColor,
 	textColor,
 	isPublished,
+	canBeUpgradedToCoreNavigationBlock,
+	upgradeToCoreNavigationBlock,
 } ) => {
+	const [ upgradeNudgeVisible, setUpgradeNudgeVisible ] = useState( true );
+
 	const { customFontSize, textAlign } = attributes;
 
 	const actualFontSize = customFontSize || fontSize.size;
 
 	return (
 		<Fragment>
+			{ upgradeNudgeVisible && canBeUpgradedToCoreNavigationBlock && (
+				<Notice
+					onRemove={ () => setUpgradeNudgeVisible( false ) }
+					actions={ [
+						{
+							label: __( 'Upgrade Block', 'full-site-editing' ),
+							onClick: upgradeToCoreNavigationBlock,
+						},
+					] }
+				>
+					<p>
+						<span className="posts-list__message">
+							{ __(
+								'An improved version of the Navigation block is available. Upgrade for a better, more natural way to manage your Navigation.',
```

Comment:
It would be good to extract the notice to a separate component / PR. It already gets duplicated [here](https://github.com/Automattic/wp-calypso/pull/38218/files#diff-c55a780f7eb5d28c236f2befda183b65R74).

Reply:
Agreed. If this is something that can be shared across FSE then that's a good idea. I'd be inclined to ship "as is" and then unify in a separate PR.